source/deprecation_policy.html.md

```diff
@@ -0,0 +1,133 @@
+
+For example: If API version v2 is released on March 25th, 2019, and API version v3 is released August 7th, 2019, then v2 would be discontinued no earlier than November 7th, 2019, 3 months after the release of v3.
+
+After the deprecation period has elapsed, requests made with the discontinued version number to endpoints which remain unchanged will be routed internally by Welkin to the oldest active version. Requests made with the discontinued version number to endpoints which have changed will return a 400 error indicating that the endpoint has changed and the request is invalid.
+
+Below is an example release timeline:
+
+![api_versions_deprecation_timeline_121019](api_versions_deprecation_timeline_121019.png)
+
+Developers can find specific information about our versions, non-backwards-compatible changes, and release dates on our [changelog page](/changelog.html).
+
+# How will Welkin notify developers when an API version is deprecated and discontinued?
+
+Welkin will use commercially reasonable methods to notify developers using the Data API of the following: new API versions, deprecations of API versions, and discontinuations of API version.
+
+These notifications will be sent via the developers listserve, posts on Welkin's knowledge base, and programmatically via API response headers. Welkin's Customer Success Managers will also communicate with customers to inform them as to which changes are likely to affect their implementations of Welkin.
+
+Emails will be sent to the developers listserve when a new API version is released (thus deprecating the old version) and when an API version is going to be discontinued.
+
+All API responses will contain a `x-current-api-version` header which will be set to the current version of the Data API, a `x-api-deprecated` header when calling any deprecated API versions, and a `x-api-discontinued` header when calling any discontinued API versions.
+
+# Definitions
+
+Terms used in this policy have specific meaning. They are defined below.
+
+### Version
+
+This refers to the version number found in the URL path of any API request, for example: `GET /v{version_number}/patients`.  This style of versioning is known as URI versioning.
+
+### Non-backwards-compatible changes
+
+Non-backwards-compatible changes are any changes that require developers to make changes to maintain existing functionality. Breaking changes include things like:
+
+* Removing a parameter from a request or response schema
+* Modifying the name for any parameters or endpoints
+* Change in the type of a parameter
+* Adding a required parameter
+* Changing the semantics of a parameter value
+* Changes to error codes or error messages to existing endpoints
```

Comment:
Does this mean we're committing to maintain current error codes / behaviour within an API version? That's not really documented right now.

I don't disagree that this might be breaking, just flagging that it might feel constraining.

Reply:
i think error codes are part of the spec and thus breaking